### PR TITLE
Updated and Created CDL blocks

### LINKED ISSUE #2865: Modify CDL blocks based on ASHRAE 231P discussion
This issue is to update/add existing/new blocks in `Buildings.Controls.OBC.CDL` package, based on the discussion in ASHRAE 231P project.

1. Existing blocks to be updated:
    - [x] `CDL.Continuous.Add` and `CDL.Integers.Add`: do not have a gain factor for inputs, only add two input values 

2. Existing blocks to be renamed:
    - [x] change `CDL.Continuous.Division` to `CDL.Continuous.Divide`
    - [x] change `CDL.Continuous.Gain` to `CDL.Continuous.MultiplyByParameter`
    - [x] change `CDL.Continuous.MovingMean` to `CDL.Continuous.MovingAverage`
    - [x] change `CDL.Continuous.Product` to `CDL.Continuous.Multiply`
    - [x] change `CDL.Integers.Product` to `CDL.Integers.Multiply`
    - [x] change `CDL.Continuous.SlewRateLimiter` to `CDL.Continuous.LimitSlewRate`

3. New blocks to be added:
    - [x] `CDL.Continuous.Acos`
    - [x] `CDL.Continuous.Asin`
    - [x] `CDL.Continuous.Subtract`
    - [x] `CDL.Integers.Subtract`

4. Move existing blocks:
    - [x] move `CDL.Continuous.Feedback` to folder `Buildings.Obsolete.Controls.OBC.CDL.Continuous`
    - [x] move the old `CDL.Continuous.Add` and `CDL.Integers.Add` to folder `Buildings.Obsolete.Controls.OBC.CDL.Continuous` and `Buildings.Obsolete.Controls.OBC.CDL.Integers`

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #2865.

- [x] added new CDL blocks, renamed existing blocks, and moved blocks to <code>Obsolete</code> package, see #2865.
- [x] renamed Dymola command script (.mos) file, renamed the reference results and the OpenModelica script 
- [x] created Dymola conversion script
- [x] updated release note

### CODE REVIEW

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/CoolingCoil.mo [hunk 1]
```diff
@@ -5,7 +5,7 @@ model CoolingCoil "Controller for cooling coil valve"
     Buildings.Controls.OBC.CDL.Types.SimpleController.PI
     "Type of controller"
     annotation(Dialog(group="Cooling coil loop signal"));
-  parameter Real kCooCoi(final unit="1/K")=0.1
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/CoolingCoil.mo:8]: Should this have unit "1/K"?

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Controller.mo [hunk 2]
```diff
@@ -73,7 +73,7 @@ block Controller
     Buildings.Controls.OBC.CDL.Types.SimpleController.PI
     "Type of controller"
     annotation(Dialog(group="Cooling coil control"));
-  parameter Real kCooCoi(final unit="1/K")=0.1
```
Comment (mwetter) [inline Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Controller.mo:76]: Should this have unit "1/K"?

File: Buildings/Controls/OBC/Utilities/SetPoints/SupplyReturnTemperatureReset.mo [hunk 3]
```diff
@@ -109,76 +109,51 @@ protected
...
-    "Relative heating load = Q_flow/Q_flow_nominal"
+    annotation (Placement(transformation(extent={{-40,-52},{-20,-32}})));
+  CDL.Continuous.Divide qRel "Relative heating load = Q_flow/Q_flow_nominal"
     annotation (Placement(transformation(extent={{0,24},{20,44}})));
-  CDL.Continuous.Add dTFlu2_nom(
-    final k1=0.5,
-    final k2=-0.5,
-    y(final quantity="TemperatureDifference",
-      final unit="K"))
+  CDL.Continuous.Subtract dTFlu2_nom
```
Comment (mwetter) [inline Buildings/Controls/OBC/Utilities/SetPoints/SupplyReturnTemperatureReset.mo:131]: Units should be put back.

**Review by mwetter — COMMENTED** [review]
@JayHuLBL 
Can you please address the two inline comments. Then this is ready to merge when the tests pass.

File: Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Controller.mo (hunk 2 above)
Comment (JayHuLBL) [reply Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Controller.mo:76]: After adding the unit, when checking the `SingleZone.VAV.Controller` model, I cannot see any warning or error. However, when checking the validation model `SingleZone.VAV.Validation.Controller`, I see following error:
```
Incompatible units.

Incompatible units in
der(conVAV.cooCoi.cooCoiPI.I.y) = conVAV.cooCoi.cooCoiPI.I.k*conVAV.cooCoi.cooCoiPI.I.u;

The unit issue reported appeared in the equation
der(conVAV.cooCoi.cooCoiPI.I.y) = conVAV.cooCoi.cooCoiPI.I.k*conVAV.cooCoi.cooCoiPI.I.u;
Found in class Buildings.Controls.OBC.CDL.Continuous.IntegratorWithReset, /home/jianjun/GitFolder/modelica-buildings/Buildings/Controls/OBC/CDL/Continuous/IntegratorWithReset.mo at line 27, and used in component conVAV.cooCoi.cooCoiPI.I.
```

I did add the unit for the constant source output in commit 3271866. But still see the error.

Comment (mwetter) [reply Buildings/Controls/OBC/ASHRAE/G36_PR1/AHUs/SingleZone/VAV/Controller.mo]: I see, the mistake was in `IntegratorWithReset`, which I fixed in 1fc9b22a840b915fba48a335d5b3d56a8a46e1a6. In that commit, I also added the units back for `kCooCoi`.

**Comment by mwetter** [comment]
@JayHuLBL : FYI, I am working on resolving the failed test which needs a patch in BuildingsPy

**Pull request merged**